### PR TITLE
[scope-guard] Add new port

### PR DESCRIPTION
**Pull request by oold**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/scope-guard/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
```
Comment (FrankXie05) [inline ports/scope-guard/portfile.cmake:1]:
```suggestion
# Header-only library
vcpkg_from_github(

```
We can note this. :)

File: ports/scope-guard/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ricab/scope_guard
+    REF v1.0.0
```
Comment (FrankXie05) [inline ports/scope-guard/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"

```

File: ports/scope-guard/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ricab/scope_guard
+    REF v1.0.0
+    SHA512 48d75658c9f2c8770b6379f968ee498bc90b5e5e14f26c6e89ffd7d50a8230260db262da27fe4aa3c6c1b302fe73d9aa9a7c853e01adcf5a6313e3462d3a8407
+    HEAD_REF main
+)
+
+file(INSTALL "${SOURCE_PATH}/scope_guard.hpp" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (FrankXie05) [inline ports/scope-guard/portfile.cmake:10]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

```
We can use the new function.

File: ports/scope-guard/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "scope-guard",
+  "version-semver": "1.0.0",
```
Comment (FrankXie05) [inline ports/scope-guard/vcpkg.json:3]: Is semer the semantic rules required upstream? If not, we should change to the `"version": "1.0.0",`

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/scope-guard/vcpkg.json (hunk 4 above)
Comment (oold) [reply ports/scope-guard/vcpkg.json:3]: Upstream uses semver.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**